### PR TITLE
Updated AppTP lifecycle checks

### PR DESCRIPTION
**Pull request by samsymons**
Task/Issue URL: https://app.asana.com/0/414235014887631/1204492688830643/f
Tech Design URL:
CC: @SlayterDev 

**Description**:

This PR adds additional AppTP checks. See the parent Asana task for more info.

**Steps to test this PR**:
1. Check that AppTP runs as expected

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: PacketTunnelProvider/AppTrackingProtectionPacketTunnelProvider.swift [hunk 1]
```diff
@@ -98,11 +109,12 @@ class AppTrackingProtectionPacketTunnelProvider: NEPacketTunnelProvider {
     }
     
     override func sleep(completionHandler: @escaping () -> Void) {
-        // Add code here to get ready to sleep.
-        completionHandler()
+        Pixel.fire(pixel: .appTPVPNWake) { _ in
+            completionHandler()
+        }
     }
```
Comment (samsymons) [inline PacketTunnelProvider/AppTrackingProtectionPacketTunnelProvider.swift]: I'm not sure that this function is being called as we don't have the appropriate flag set, but I'd like to be sure about that, so let's monitor it.

File: PacketTunnelProvider/AppTrackingProtectionPacketTunnelProvider.swift [hunk 2]
```diff
@@ -98,11 +109,12 @@ class AppTrackingProtectionPacketTunnelProvider: NEPacketTunnelProvider {
...
     
     override func sleep(completionHandler: @escaping () -> Void) {
-        // Add code here to get ready to sleep.
-        completionHandler()
+        Pixel.fire(pixel: .appTPVPNWake) { _ in
```
Comment (SlayterDev) [inline PacketTunnelProvider/AppTrackingProtectionPacketTunnelProvider.swift]: This should be the sleep pixel right?

Comment (samsymons) [reply PacketTunnelProvider/AppTrackingProtectionPacketTunnelProvider.swift]: Indeed it should

**Review by SlayterDev — APPROVED** [review]
LGTM!

**Pull request merged**